ArchiSteamFarm.OfficialPlugins.Monitoring/MonitoringPlugin.cs

```diff
@@ -0,0 +1,203 @@
+using System.ComponentModel.DataAnnotations;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Metrics;
+using System.Globalization;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using ArchiSteamFarm.Core;
+using ArchiSteamFarm.IPC.Integration;
+using ArchiSteamFarm.OfficialPlugins.Monitoring.Localization;
+using ArchiSteamFarm.Plugins;
+using ArchiSteamFarm.Plugins.Interfaces;
+using ArchiSteamFarm.Steam;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Metrics;
+using SteamKit2;
+
+namespace ArchiSteamFarm.OfficialPlugins.Monitoring;
+
+[Export(typeof(IPlugin))]
+[SuppressMessage("ReSharper", "MemberCanBeFileLocal")]
+internal sealed class MonitoringPlugin : OfficialPlugin, IWebServiceProvider, IGitHubPluginUpdates, IASF, IDisposable {
+	private const string MeterName = nameof(ArchiSteamFarm);
+
+	private const string MetricNamePrefix = "asf";
+
+	[JsonInclude]
+	[Required]
+	public override string Name => nameof(MonitoringPlugin);
+
+	/// <inheritdoc />
+	public string RepositoryName => SharedInfo.GithubRepo;
+
+	[JsonInclude]
+	[Required]
+	public override Version Version => typeof(MonitoringPlugin).Assembly.GetName().Version ?? throw new InvalidOperationException(nameof(Version));
+
+	private static bool Enabled => ASF.GlobalConfig?.IPC ?? false;
+
+	private Meter? Meter;
+
+	public void Dispose() => Meter?.Dispose();
+
+	public Task OnASFInit(IReadOnlyDictionary<string, JsonElement>? additionalConfigProperties = null) {
+		if (additionalConfigProperties == null) {
+			return Task.CompletedTask;
+		}
+
+		if (!Enabled) {
+			ASF.ArchiLogger.LogGenericInfo(string.Format(CultureInfo.CurrentCulture, Strings.PluginDisabledInConfig, nameof(MonitoringPlugin)));
+
+			return Task.CompletedTask;
+		}
+
+		ASF.ArchiLogger.LogGenericInfo(string.Format(CultureInfo.CurrentCulture, Strings.PluginInitializedAndEnabled, nameof(MonitoringPlugin)));
+
+		return Task.CompletedTask;
+	}
+
+	public void OnConfiguringEndpoints(IApplicationBuilder app) {
+		ArgumentNullException.ThrowIfNull(app);
+
+		if (!Enabled) {
+			return;
+		}
+
+		app.UseEndpoints(static builder => builder.MapPrometheusScrapingEndpoint());
+	}
+
+	public void OnConfiguringServices(IServiceCollection services) {
+		ArgumentNullException.ThrowIfNull(services);
+
+		if (!Enabled) {
+			return;
+		}
+
+		services.AddOpenTelemetry().WithMetrics(
+			static builder => {
+				builder.AddPrometheusExporter(static config => config.ScrapeEndpointPath = "/Api/metrics");
+				builder.AddRuntimeInstrumentation();
+				builder.AddAspNetCoreInstrumentation();
+				builder.AddHttpClientInstrumentation();
+				builder.AddMeter(MeterName);
+			}
+		);
+	}
+
+	public override Task OnLoaded() {
+		Utilities.WarnAboutIncompleteTranslation(Strings.ResourceManager);
+
+		Meter = new Meter(MeterName, Version.ToString());
```

Comment:
Would be better to move it outside of `OnLoaded()` e.g. to configuring endpoints, so we could avoid pointless memory allocation when plugin is not enabled.

You can check against `Meter` being null, there is an edge case when configuring endpoints and services can be called more than once - IPC restart, triggered by `.config` edit. Best therefore would be to just check if meter is null and skip its initialization, otherwise initialize it. You can extract it to some nice `InitializeMeter()` function and just call it with return inside if meter not null.